client/src/selectors/schedules.js

```diff
@@ -0,0 +1,24 @@
+export default (schedules, { startTime, endTime }) => {
+  const filteredItems = [];
+  schedules.map(schedule =>
+    schedule.timeSegments.map((timeSegment) => {
+      if (
+        (timeSegment.startTime === 0 || timeSegment.startTime >= startTime) &&
+        (timeSegment.endTime === 2147483647 || timeSegment.endTime <= endTime)
```

Comment:
please use the constants in place of `0` and `2147483647`. I'm also pretty sure you shouldn't need to special-case these but I can't think of the logic right now.